### PR TITLE
fix(definition-list): fix dl focused elements outline cut by "overflow: hidden" - FE-3211

### PR DESCRIPTION
**Pull request by mmaciorowski-sage**
### Proposed behaviour
Remove `overflow: hidden` from `definition-list` component.
Zrzut ekranu 2021-04-26 o 10 23 03

### Current behaviour
Zrzut ekranu 2021-04-26 o 10 23 11

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### Testing instructions
https://codesandbox.io/s/carbon-quickstart-forked-30grt

### CODE REVIEW

File: src/components/definition-list/dl.component.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ const Dl = ({
   w = 50,
   dtTextAlign = "right",
   ddTextAlign = "left",
+  pl = "3px",
```
Comment (edleeks87) [inline src/components/definition-list/dl.component.js:13]: Can you add these default values to the tests please https://github.com/Sage/carbon/blob/4e8087fc78051e8b2bd011d9283be3143adf35d4/src/components/definition-list/definition-list.spec.js#L34

Comment (mmaciorowski-sage) [reply src/components/definition-list/dl.component.js:13]: good point, thanks, added

File: src/components/definition-list/dl.component.js [hunk 2]
```diff
@@ -57,7 +61,15 @@ const Dl = ({
   };
 
   return (
-    <StyledDl w={w} data-component="dl" {...props}>
+    <StyledDl
```
Comment (dawidzarzycki) [inline src/components/definition-list/dl.component.js:64]: you can set 
```js
px={'3px'}
mx={'-3px'}
```

**Comment by dawidzarzycki** [comment]
@mmaciorowski-sage Hey!
Maybe better solution for this will be to remove `overflow: hidden` property from `dl` component instead of adding paddings and margins

File: src/components/definition-list/dl.component.js (hunk 2 above)
Comment (mmaciorowski-sage) [reply src/components/definition-list/dl.component.js]: good point, thanks, changed

**Comment by mmaciorowski-sage** [comment]
> @mmaciorowski-sage Hey!
> Maybe better solution for this will be to remove `overflow: hidden` property from `dl` component instead of adding paddings and margins

I talked to Daniel and he told me that it was because we wanted the list to fit in to the size of the `Tile` for example. So if someone had a definition-list with 50 lines, it wouldn’t overflow the size of the `Tile`.

**Review by dawidzarzycki — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Comment by mmaciorowski-sage** [comment]
> @mmaciorowski-sage Hey!
> Maybe better solution for this will be to remove `overflow: hidden` property from `dl` component instead of adding paddings and margins

FYI At first I assumed it was necessary, but it turned out that it could be removed.

**Review by ZhuoyuJin — APPROVED** [review]
Tested in Edge, Chrome, Firefox, and Safari. No issues found, testing passed.

**Pull request merged**